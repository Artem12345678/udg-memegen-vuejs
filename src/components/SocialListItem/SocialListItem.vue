<template>
  <a class="social__link" :href="url" target="_blank">
    <img :src="image">
  </a>
</template>

<script>
import instagram from '../../assets/instagram.svg';
import github from '../../assets/github.svg';

export default {
  name: 'SociaItem',
  props: ['url', 'icon'],
  computed: {
    image() {
      return this.icon === 'instagram' ? instagram : github;
    },
  },
};
</script>

<style lang="scss" scoped>
  .social__link {
    display: flex;
    width: 28px;
    height: 28px;
    transition: border-color .25s ease-in-out;
    text-align: center;
    border: 1px solid transparent;
    border-radius: 50%;
    align-items: center;
    justify-content: center;

    &:focus,
    &:active,
    &:hover {
      border-color: #fff;
    }
  }
</style>
