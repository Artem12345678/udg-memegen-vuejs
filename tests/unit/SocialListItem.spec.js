import { shallowMount } from '@vue/test-utils';

import SocialListItem from '@/components/SocialListItem/SocialListItem.vue';

describe('SocialListItem.vue', () => {
  let wrapper;

  const links = [
    {
      url: 'https://www.github.com/Artem12345678',
      icon: 'github',
    },
    {
      url: 'https://www.instagram.com/ystimenkoartem',
      icon: 'instagram',
    },
  ];

  it('matches snapshot', () => {
    wrapper = shallowMount(SocialListItem, {
      propsData: {
        url: links[0].url,
        icon: links[0].icon,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders Github icon', () => {
    wrapper = shallowMount(SocialListItem, {
      propsData: {
        url: links[0].url,
        icon: links[0].icon,
      },
    });

    expect(wrapper.find('.social__link').exists()).toBe(true);
    expect(wrapper.find('.social__link').attributes('href')).toBe(links[0].url);
    expect(wrapper.find('.social__link img').attributes('src')).toBe(
      SocialListItem.computed.image(),
    );
  });

  it('renders Instagram icon', () => {
    wrapper = shallowMount(SocialListItem, {
      propsData: {
        url: links[1].url,
        icon: links[1].icon,
      },
    });

    expect(wrapper.find('.social__link').exists()).toBe(true);
    expect(wrapper.find('.social__link').attributes('href')).toBe(links[1].url);
    expect(wrapper.find('.social__link img').attributes('src')).toBe(
      SocialListItem.computed.image(),
    );
  });
});
