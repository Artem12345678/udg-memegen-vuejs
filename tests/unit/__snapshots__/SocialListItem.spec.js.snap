// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialListItem.vue matches snapshot 1`] = `
<a
  class="social__link"
  href="https://www.github.com/Artem12345678"
  target="_blank"
>
  <img
    src=""
  />
</a>
`;
